justfile

```diff
@@ -7,5 +7,15 @@ default:
 build:
     docker build . -t research-template
 
-smoke-test:
-    docker run --rm research-template ls /opt/venv/bin/python3.10
\ No newline at end of file
+check-image-exists:
+    # Extra brackets are for Just escaping.
+    docker image inspect --format='{{{{.Id}}}}' research-template
+
+test-smoke: check-image-exists
+    docker run --rm research-template ls /opt/venv/bin/python3.10
```

Comment:
This is probably less necessary now than it was.

We didn't have anything else before, so having some rudimentary check was better than nothing.

It might still be useful to have something that can fail fast, so the CI fails quickly before the CI uploads the Docker image (which is a slow step). I'm not sure what that check might be though.

Reply:
Have moved this to a check of the Python 3.10 install in the test step.